qiskit/transpiler/passes/basis/basis_translator.py

```diff
@@ -295,6 +284,43 @@ def replace_node(node, instr_map):
 
         return dag
 
+    def _update_basis(self, source_basis, dag):
+        for node in dag.op_nodes():
+            if not dag.has_calibration_for(node):
+                source_basis.add((node.name, node.op.num_qubits))
+            if isinstance(node.op, ControlFlowOp):
+                for block in node.op.blocks:
+                    block_dag = circuit_to_dag(block)
+                    self._update_basis(source_basis, block_dag)
+
+    def _update_basis_target(self, source_basis, qargs_local_source_basis, qarg_indices, dag):
```

Comment:
This could also be made to be a "constructor" rather than "mutator" function (since it's logically a constructor).